packages/web/pages/index.tsx

```diff
@@ -1,18 +1,92 @@
-import { observer } from "mobx-react-lite";
 import type { NextPage } from "next";
-import { ProgressiveSvgImage } from "../components/progressive-svg-image";
+import { observer } from "mobx-react-lite";
+import { useStore } from "../stores";
 import { TradeClipboard } from "../components/trade-clipboard";
+import { ProgressiveSvgImage } from "../components/progressive-svg-image";
+import { useEffect, useMemo, useRef, useState } from "react";
 
 const Home: NextPage = observer(function () {
+  const { chainStore, queriesStore } = useStore();
+
+  const containerRef = useRef<HTMLElement | null>(null);
+  const clipboardRef = useRef<HTMLDivElement | null>(null);
+  const [containerSize, setContainerSize] = useState<
+    | {
+        width: number;
+        height: number;
+      }
+    | undefined
+  >(undefined);
+  const [clipboardWidth, setClipboardWidth] = useState<number | undefined>(
+    undefined
+  );
+
+  useEffect(() => {
```

Comment:
In the `mobile-foundation` branch, I created a `useWindowSize()` hook that may serve the needs we have here. Perhaps add a TODO to utilize that instead.